### PR TITLE
fix: tuple endpoint return invalid error

### PR DESCRIPTION
**Pull request by aiwantaozi**
**Problem:**
Tuple endpoint returns an invalid error

**Solution:**
Should not return invalid error while 

**Related Issue:**

### CODE REVIEW

File: pkg/terraform/parser/parser.go [hunk 1]
```diff
@@ -225,15 +225,17 @@ func ParseStateOutput(revision *model.ResourceRevision) ([]types.OutputValue, er
 				val = []byte(`"<sensitive>"`)
 			}
 
+			s := translator.SchemaOfType(
```
Comment (alexcodelf) [inline pkg/terraform/parser/parser.go:228]: What is the different of this change between before and after?

**Review by thxCode — APPROVED** [review]
LGTM

File: pkg/apis/resource/extension.go [hunk 2]
```diff
@@ -304,10 +304,20 @@ func (h Handler) getEndpointsFromOutput(ctx context.Context, id object.ID) ([]Ac
 				Name:      v.Name,
 			})
 		case v.Schema.Type == openapi3.TypeArray:
-			if v.Schema.Items == nil || v.Schema.Items.Value == nil ||
-				v.Schema.Items.Value.Type != openapi3.TypeString {
-				// For list and set: all elements are the same type.
-				return nil, invalidTypeErr
+			if v.Schema.Items != nil && v.Schema.Items.Value != nil {
+				// Array.
+				if v.Schema.Items.Value.Type != "" && v.Schema.Items.Value.Type != openapi3.TypeString {
```
Comment (alexcodelf) [inline pkg/apis/resource/extension.go:309]: When `v.Schema.Items.Value.Type != openapi3.TypeString`, it must not equal to `""`, So it just need to change to 
```
if v.Schema.Items.Value.Type != openapi3.TypeString {
    xxx
}
```

File: pkg/terraform/parser/parser.go (hunk 1 above)
Comment (aiwantaozi) [reply pkg/terraform/parser/parser.go:228]: just makes the code read easier

File: pkg/apis/resource/extension.go (hunk 2 above)
Comment (aiwantaozi) [reply pkg/apis/resource/extension.go:309]: when the type isn't set, also allows to pass

**Pull request merged**